spec/models/info_request_spec.rb

```diff
@@ -50,13 +50,254 @@
+      info_request = FactoryGirl.create(:info_request)
+      email, raw_email = email_and_raw_email
+      info_request.receive(email, raw_email)
+      expect(info_request.incoming_messages.first.raw_email.data).
+        to eq(raw_email)
+      expect(info_request.incoming_messages.first.raw_email).to be_persisted
+    end
+
+    it 'marks the request as awaiting description' do
+      info_request = FactoryGirl.create(:info_request)
+      email, raw_email = email_and_raw_email
+      info_request.receive(email, raw_email)
+      expect(info_request.awaiting_description).to be true
+    end
+
+    it 'logs an event' do
+      info_request = FactoryGirl.create(:info_request)
+      email, raw_email = email_and_raw_email
+      info_request.receive(email, raw_email)
+      expect(info_request.info_request_events.last.incoming_message.id).
+        to eq(info_request.incoming_messages.last.id)
+      expect(info_request.info_request_events.last).to be_response
+    end
+
+    it 'accepts a rejected reason' do
+      info_request = FactoryGirl.create(:info_request)
+      email, raw_email = email_and_raw_email
+      info_request.receive(email, raw_email, false, 'rejected for testing')
+      expect(info_request.info_request_events.last.params[:rejected_reason]).
+        to eq('rejected for testing')
+    end
+
+    context 'notifying the request owner' do
+
+      it 'notifies the user that a response has been received' do
+        info_request = FactoryGirl.create(:info_request)
+        email, raw_email = email_and_raw_email
+        info_request.receive(email, raw_email)
+        notification = ActionMailer::Base.deliveries.last
+        expect(notification.to).to include(info_request.user.email)
+        ActionMailer::Base.deliveries.clear
+      end
+
+      it 'does not notify when the request is external' do
+        info_request = FactoryGirl.create(:external_request)
+        email, raw_email = email_and_raw_email
+        info_request.receive(email, raw_email)
+        expect(ActionMailer::Base.deliveries).to be_empty
+        ActionMailer::Base.deliveries.clear
+      end
+
+    end
+
+    context 'allowing new responses' do
+
+      it 'from nobody' do
+        attrs = { :allow_new_responses_from => 'nobody',
+                  :handle_rejected_responses => 'holding_pen' }
+        info_request = FactoryGirl.create(:info_request, attrs)
+        email, raw_email = email_and_raw_email
+        info_request.receive(email, raw_email)
+        holding_pen = InfoRequest.holding_pen_request
+        msg = 'This request has been set by an administrator to "allow new ' \
+              'responses from nobody"'
+        expect(holding_pen.incoming_messages.size).to eq(1)
+        expect(holding_pen.info_request_events.last.params[:rejected_reason]).
+          to eq(msg)
+      end
+
+      it 'from anybody' do
+        attrs = { :allow_new_responses_from => 'anybody',
+                  :handle_rejected_responses => 'holding_pen' }
```

Comment:
Use the new Ruby 1.9 hash syntax.<br>Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.